src/softmax.jl

```diff
@@ -17,32 +17,12 @@ independent.
       0.244728
       0.665241
 """
-softmax(xs) = softmax!(similar(xs), xs)
+softmax(xs; dims=1) = softmax!(similar(xs), xs, dims)
```

Comment:
Instead of forwarding to `softmax!`, can you just write an out-of-place version here? That'll fix the gradient issue.